New resource: aws_cloudwatch_event_archive

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

Looks like there's now an API to enable long-term archival of eventbridge events which can later be replayed: 

* https://aws.amazon.com/blogs/aws/new-archive-and-replay-events-with-amazon-eventbridge/
* https://aws.amazon.com/about-aws/whats-new/2020/11/amazon-eventbridge-introduces-support-for-event-replay/

```
client.create_archive(
    ArchiveName='string',
    EventSourceArn='string',
    Description='string',
    EventPattern='string',
    RetentionDays=123
)
```

### New or Affected Resource(s)

<!--- Please list the new or affected resources and data sources. --->

* aws_cloudwatch_event_archive

### Potential Terraform Configuration

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
resource "aws_cloudwatch_event_bus" "messenger" {
  name = "chat-messages"
}

resource "aws_cloudwatch_event_archive" "archive" {
  archive_name = "archive"

  event_source_arn = aws_cloudwatch_event_bus.messenger.arn

  description = "Archive of chat message events"

  event_pattern = <<EOF
{
  "detail-type": [
    "customerCreated"
  ]
}
EOF

  retention_days = 0
}
```

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor blog posts or documentation? For example:

* https://aws.amazon.com/about-aws/whats-new/2018/04/introducing-amazon-ec2-fleet/
--->

* https://docs.aws.amazon.com/cli/latest/reference/events/create-archive.html
* https://awsapichanges.info/archive/changes/034487-events.html#CreateArchive

@heitorlessa 
```
Error: rpc error: code = Unavailable desc = transport is closing


panic: interface conversion: interface {} is int, not int64
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5:
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: goroutine 53 [running]:
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: github.com/terraform-providers/terraform-provider-aws/aws.buildCreateArchiveInputStruct(0xc001213780, 0xc001365770, 0x16976d0, 0xc001365710)
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-aws/aws/resource_aws_cloudwatch_event_archive.go:169 +0x507
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsCloudWatchEventArchiveCreate(0xc001213780, 0x6767340, 0xc001281600, 0x0, 0xffffffffffffffff)
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-aws/aws/resource_aws_cloudwatch_event_archive.go:67 +0x55
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc000603200, 0x7ecd800, 0xc00030d580, 0xc001213780, 0x6767340, 0xc001281600, 0x0, 0x0, 0x0)
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.3/helper/schema/resource.go:278 +0x88
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000603200, 0x7ecd800, 0xc00030d580, 0xc0009a1420, 0xc0012039c0, 0x6767340, 0xc001281600, 0x0, 0x0, 0x0, ...)
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.3/helper/schema/resource.go:396 +0x65b
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00109f620, 0x7ecd800, 0xc00030d580, 0xc0012051d0, 0xc00030d580, 0x0, 0x7ef9860)
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.3/helper/schema/grpc_provider.go:955 +0x8cf
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc0013eb8a0, 0x7ecd800, 0xc00030d580, 0xc0009a11f0, 0xc0013eb8a0, 0xc001232450, 0xc00065aba0)
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.2.1/tfprotov5/server/server.go:332 +0xae
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x725dc40, 0xc0013eb8a0, 0x7ecd8c0, 0xc001232450, 0xc000f8d440, 0x0, 0x7ecd8c0, 0xc001232450, 0xc001226300, 0x17a)
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.2.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: google.golang.org/grpc.(*Server).processUnaryRPC(0xc00001f340, 0x7ef0ee0, 0xc00135c180, 0xc001224100, 0xc000338c90, 0xb31cb60, 0x0, 0x0, 0x0)
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:1194 +0x522
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: google.golang.org/grpc.(*Server).handleStream(0xc00001f340, 0x7ef0ee0, 0xc00135c180, 0xc001224100, 0x0)
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:1517 +0xd05
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000bef3d0, 0xc00001f340, 0x7ef0ee0, 0xc00135c180, 0xc001224100)
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:859 +0xa5
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: created by google.golang.org/grpc.(*Server).serveStreams.func1
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd
2021/02/22 18:47:03 [DEBUG] module.cazoo_events_forwarder.aws_cloudwatch_event_archive.default: apply errored, but we're indicating that via the Error pointer rather than returning it: rpc error: code = Unavailable desc = transport is closing
2021-02-22T18:47:03.205Z [WARN]  plugin.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021-02-22T18:47:03.205Z [DEBUG] plugin: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/aws/3.29.0/darwin_amd64/terraform-provider-aws_v3.29.0_x5 pid=79275 error="exit status 2"
2021/02/22 18:47:03 [TRACE] EvalMaybeTainted: module.cazoo_events_forwarder.aws_cloudwatch_event_archive.default encountered an error during creation, so it is now marked as tainted
2021/02/22 18:47:03 [TRACE] EvalWriteState: removing state object for module.cazoo_events_forwarder.aws_cloudwatch_event_archive.default
2021/02/22 18:47:03 [TRACE] EvalApplyProvisioners: aws_cloudwatch_event_archive.default has no state, so skipping provisioners
2021/02/22 18:47:03 [TRACE] EvalMaybeTainted: module.cazoo_events_forwarder.aws_cloudwatch_event_archive.default encountered an error during creation, so it is now marked as tainted
2021/02/22 18:47:03 [TRACE] EvalWriteState: removing state object for module.cazoo_events_forwarder.aws_cloudwatch_event_archive.default
2021/02/22 18:47:03 [TRACE] vertex "module.cazoo_events_forwarder.aws_cloudwatch_event_archive.default": visit complete
2021/02/22 18:47:03 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/hashicorp/aws\"] (close)" errored, so skipping
2021/02/22 18:47:03 [TRACE] dag/walk: upstream of "module.cazoo_events_forwarder (close)" errored, so skipping
2021/02/22 18:47:03 [TRACE] dag/walk: upstream of "meta.count-boundary (EachMode fixup)" errored, so skipping
2021/02/22 18:47:03 [TRACE] dag/walk: upstream of "root" errored, so skipping
2021-02-22T18:47:03.235Z [DEBUG] plugin: plugin exited



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

SECURITY WARNING: the "crash.log" file that was created may contain
sensitive information that must be redacted before it is safe to share
on the issue tracker.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

I'm going to try upgrading from `v0.14.2`to v0.14.7 and see if I have any luck

```
panic: interface conversion: interface {} is int, not int64
2021-02-22T18:47:03.200Z [DEBUG] plugin.terraform-provider-aws_v3.29.0_x5:
```

Might have something todo with this ☝️

--- a/.changelog/17754.txt
+++ b/.changelog/17754.txt
@@ -1,0 +1,3 @@
+``release-note:bug
+resource/aws_cloudwatch_event_archive: Fix retention_days type conversion on creation
+```

--- a/aws/resource_aws_cloudwatch_event_archive.go
+++ b/aws/resource_aws_cloudwatch_event_archive.go
@@ -92,6 +92,12 @@ func resourceAwsCloudWatchEventArchiveRead(d *schema.ResourceData, meta interfac
 
 	out, err := conn.DescribeArchive(input)
 
+	if isAWSErr(err, events.ErrCodeResourceNotFoundException, "") {
+		log.Printf("[WARN] CloudWatch Events archive (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
 	if err != nil {
 		return fmt.Errorf("Error reading CloudWatch Events archive: %w", err)
 	}

--- a/aws/resource_aws_cloudwatch_event_archive.go
+++ b/aws/resource_aws_cloudwatch_event_archive.go
@@ -166,8 +166,7 @@ func buildCreateArchiveInputStruct(d *schema.ResourceData) (*events.CreateArchiv
 	}
 
 	if v, ok := d.GetOk("retention_days"); ok {
-		retentionInDays := int64(v.(int))
-		input.RetentionDays = aws.Int64(retentionInDays)
+		input.RetentionDays = aws.Int64(int64(v.(int)))
 	}
 
 	return &input, nil
@@ -191,8 +190,7 @@ func buildUpdateArchiveInputStruct(d *schema.ResourceData) (*events.UpdateArchiv
 	}
 
 	if v, ok := d.GetOk("retention_days"); ok {
-		retentionInDays := int64(v.(int))
-		input.RetentionDays = aws.Int64(retentionInDays)
+		input.RetentionDays = aws.Int64(int64(v.(int)))
 	}
 
 	return &input, nil

--- a/aws/resource_aws_cloudwatch_event_archive.go
+++ b/aws/resource_aws_cloudwatch_event_archive.go
@@ -166,7 +166,8 @@ func buildCreateArchiveInputStruct(d *schema.ResourceData) (*events.CreateArchiv
 	}
 
 	if v, ok := d.GetOk("retention_days"); ok {
-		input.RetentionDays = aws.Int64(v.(int64))
+		retentionInDays := int64(v.(int))
+		input.RetentionDays = aws.Int64(retentionInDays)
 	}
 
 	return &input, nil
@@ -196,5 +197,3 @@ func buildUpdateArchiveInputStruct(d *schema.ResourceData) (*events.UpdateArchiv
 
 	return &input, nil
 }
-
-// create a datasource

--- a/aws/resource_aws_cloudwatch_event_archive_test.go
+++ b/aws/resource_aws_cloudwatch_event_archive_test.go
@@ -143,7 +143,7 @@ func TestAccAWSCloudWatchEventArchive_disappears(t *testing.T) {
 				Config: testAccAWSCloudWatchEventArchiveConfig(archiveName),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckCloudWatchEventArchiveExists(resourceName, &v),
-					//testAccCheckResourceDisappears(testAccProvider, resourceAwsCloudWatchEventArchive(), resourceName),
+					testAccCheckResourceDisappears(testAccProvider, resourceAwsCloudWatchEventArchive(), resourceName),
 				),
 				ExpectNonEmptyPlan: true,
 			},

--- a/aws/resource_aws_cloudwatch_event_archive_test.go
+++ b/aws/resource_aws_cloudwatch_event_archive_test.go
@@ -70,7 +70,7 @@ func testSweepCloudWatchEventArchives(region string) error {
 	return nil
 }
 
-func TestAccAWSCloudWatchArchive_basic(t *testing.T) {
+func TestAccAWSCloudWatchEventArchive_basic(t *testing.T) {
 	var v1 events.DescribeArchiveOutput
 	archiveName := acctest.RandomWithPrefix("tf-acc-test")
 	resourceName := "aws_cloudwatch_event_archive.test"
@@ -78,12 +78,12 @@ func TestAccAWSCloudWatchArchive_basic(t *testing.T) {
 	resource.ParallelTest(t, resource.TestCase{
 		PreCheck:     func() { testAccPreCheck(t) },
 		Providers:    testAccProviders,
-		CheckDestroy: testAccCheckAWSCloudWatchArchiveDestroy,
+		CheckDestroy: testAccCheckAWSCloudWatchEventArchiveDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccAWSCloudWatchArchiveConfig(archiveName),
+				Config: testAccAWSCloudWatchEventArchiveConfig(archiveName),
 				Check: resource.ComposeTestCheckFunc(
-					testAccCheckCloudWatchArchiveExists(resourceName, &v1),
+					testAccCheckCloudWatchEventArchiveExists(resourceName, &v1),
 					resource.TestCheckResourceAttr(resourceName, "name", archiveName),
 					resource.TestCheckResourceAttr(resourceName, "retention_days", "0"),
 					testAccCheckResourceAttrRegionalARN(resourceName, "arn", "events", fmt.Sprintf("archive/%s", archiveName)),
@@ -100,7 +100,7 @@ func TestAccAWSCloudWatchArchive_basic(t *testing.T) {
 	})
 }
 
-func TestAccAWSCloudWatchArchive_update(t *testing.T) {
+func TestAccAWSCloudWatchEventArchive_update(t *testing.T) {
 	var v1 events.DescribeArchiveOutput
 	resourceName := "aws_cloudwatch_event_archive.test"
 	archiveName := acctest.RandomWithPrefix("tf-acc-test")
@@ -108,18 +108,18 @@ func TestAccAWSCloudWatchArchive_update(t *testing.T) {
 	resource.ParallelTest(t, resource.TestCase{
 		PreCheck:     func() { testAccPreCheck(t) },
 		Providers:    testAccProviders,
-		CheckDestroy: testAccCheckAWSCloudWatchArchiveDestroy,
+		CheckDestroy: testAccCheckAWSCloudWatchEventArchiveDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccAWSCloudWatchArchiveConfig(archiveName),
+				Config: testAccAWSCloudWatchEventArchiveConfig(archiveName),
 				Check: resource.ComposeTestCheckFunc(
-					testAccCheckCloudWatchArchiveExists(resourceName, &v1),
+					testAccCheckCloudWatchEventArchiveExists(resourceName, &v1),
 				),
 			},
 			{
-				Config: testAccAWSCloudWatchArchiveConfig_updateAttributes(archiveName),
+				Config: testAccAWSCloudWatchEventArchiveConfig_updateAttributes(archiveName),
 				Check: resource.ComposeTestCheckFunc(
-					testAccCheckCloudWatchArchiveExists(resourceName, &v1),
+					testAccCheckCloudWatchEventArchiveExists(resourceName, &v1),
 					resource.TestCheckResourceAttr(resourceName, "retention_days", "7"),
 					testAccCheckResourceAttrEquivalentJSON(resourceName, "event_pattern", "{\"source\":[\"company.team.service\"]}"),
 					resource.TestCheckResourceAttr(resourceName, "description", "test"),
@@ -137,13 +137,13 @@ func TestAccAWSCloudWatchEventArchive_disappears(t *testing.T) {
 	resource.ParallelTest(t, resource.TestCase{
 		PreCheck:     func() { testAccPreCheck(t) },
 		Providers:    testAccProviders,
-		CheckDestroy: testAccCheckAWSCloudWatchEventBusDestroy,
+		CheckDestroy: testAccCheckAWSCloudWatchEventArchiveDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccAWSCloudWatchArchiveConfig(archiveName),
+				Config: testAccAWSCloudWatchEventArchiveConfig(archiveName),
 				Check: resource.ComposeTestCheckFunc(
-					testAccCheckCloudWatchArchiveExists(resourceName, &v),
-					testAccCheckResourceDisappears(testAccProvider, resourceAwsCloudWatchEventArchive(), resourceName),
+					testAccCheckCloudWatchEventArchiveExists(resourceName, &v),
+					//testAccCheckResourceDisappears(testAccProvider, resourceAwsCloudWatchEventArchive(), resourceName),
 				),
 				ExpectNonEmptyPlan: true,
 			},
@@ -151,7 +151,7 @@ func TestAccAWSCloudWatchEventArchive_disappears(t *testing.T) {
 	})
 }
 
-func testAccCheckAWSCloudWatchArchiveDestroy(s *terraform.State) error {
+func testAccCheckAWSCloudWatchEventArchiveDestroy(s *terraform.State) error {
 	conn := testAccProvider.Meta().(*AWSClient).cloudwatcheventsconn
 
 	for _, rs := range s.RootModule().Resources {
@@ -173,7 +173,7 @@ func testAccCheckAWSCloudWatchArchiveDestroy(s *terraform.State) error {
 	return nil
 }
 
-func testAccCheckCloudWatchArchiveExists(n string, v *events.DescribeArchiveOutput) resource.TestCheckFunc {
+func testAccCheckCloudWatchEventArchiveExists(n string, v *events.DescribeArchiveOutput) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
 		rs, ok := s.RootModule().Resources[n]
 		if !ok {
@@ -200,7 +200,7 @@ func testAccCheckCloudWatchArchiveExists(n string, v *events.DescribeArchiveOutp
 	}
 }
 
-func TestAccAWSCloudWatchArchive_retentionSetOnCreation(t *testing.T) {
+func TestAccAWSCloudWatchEventArchive_retentionSetOnCreation(t *testing.T) {
 	var v1 events.DescribeArchiveOutput
 	archiveName := acctest.RandomWithPrefix("tf-acc-test")
 	resourceName := "aws_cloudwatch_event_archive.test"
@@ -208,12 +208,12 @@ func TestAccAWSCloudWatchArchive_retentionSetOnCreation(t *testing.T) {
 	resource.ParallelTest(t, resource.TestCase{
 		PreCheck:     func() { testAccPreCheck(t) },
 		Providers:    testAccProviders,
-		CheckDestroy: testAccCheckAWSCloudWatchArchiveDestroy,
+		CheckDestroy: testAccCheckAWSCloudWatchEventArchiveDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccAWSCloudWatchArchiveConfig_retentionOnCreation(archiveName),
+				Config: testAccAWSCloudWatchEventArchiveConfig_retentionOnCreation(archiveName),
 				Check: resource.ComposeTestCheckFunc(
-					testAccCheckCloudWatchArchiveExists(resourceName, &v1),
+					testAccCheckCloudWatchEventArchiveExists(resourceName, &v1),
 					resource.TestCheckResourceAttr(resourceName, "name", archiveName),
 					resource.TestCheckResourceAttr(resourceName, "retention_days", "1"),
 					testAccCheckResourceAttrRegionalARN(resourceName, "arn", "events", fmt.Sprintf("archive/%s", archiveName)),
@@ -230,7 +230,7 @@ func TestAccAWSCloudWatchArchive_retentionSetOnCreation(t *testing.T) {
 	})
 }
 
-func testAccAWSCloudWatchArchiveConfig(name string) string {
+func testAccAWSCloudWatchEventArchiveConfig(name string) string {
 	return fmt.Sprintf(`
 resource "aws_cloudwatch_event_bus" "test" {
   name = %[1]q
@@ -243,7 +243,7 @@ resource "aws_cloudwatch_event_archive" "test" {
 `, name)
 }
 
-func testAccAWSCloudWatchArchiveConfig_updateAttributes(name string) string {
+func testAccAWSCloudWatchEventArchiveConfig_updateAttributes(name string) string {
 	return fmt.Sprintf(`
 resource "aws_cloudwatch_event_bus" "test" {
   name = %[1]q
@@ -263,7 +263,7 @@ PATTERN
 `, name)
 }
 
-func testAccAWSCloudWatchArchiveConfig_retentionOnCreation(name string) string {
+func testAccAWSCloudWatchEventArchiveConfig_retentionOnCreation(name string) string {
 	return fmt.Sprintf(`
 resource "aws_cloudwatch_event_bus" "test" {
   name = %[1]q

--- a/aws/resource_aws_cloudwatch_event_archive_test.go
+++ b/aws/resource_aws_cloudwatch_event_archive_test.go
@@ -140,7 +140,7 @@ func TestAccAWSCloudWatchEventArchive_disappears(t *testing.T) {
 		CheckDestroy: testAccCheckAWSCloudWatchEventBusDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccAWSCloudWatchEventBusConfig(archiveName),
+				Config: testAccAWSCloudWatchArchiveConfig(archiveName),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckCloudWatchArchiveExists(resourceName, &v),
 					testAccCheckResourceDisappears(testAccProvider, resourceAwsCloudWatchEventArchive(), resourceName),

--- a/aws/resource_aws_cloudwatch_event_archive_test.go
+++ b/aws/resource_aws_cloudwatch_event_archive_test.go
@@ -200,6 +200,36 @@ func testAccCheckCloudWatchArchiveExists(n string, v *events.DescribeArchiveOutp
 	}
 }
 
+func TestAccAWSCloudWatchArchive_retentionSetOnCreation(t *testing.T) {
+	var v1 events.DescribeArchiveOutput
+	archiveName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_cloudwatch_event_archive.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSCloudWatchArchiveDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSCloudWatchArchiveConfig_retentionOnCreation(archiveName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudWatchArchiveExists(resourceName, &v1),
+					resource.TestCheckResourceAttr(resourceName, "name", archiveName),
+					resource.TestCheckResourceAttr(resourceName, "retention_days", "1"),
+					testAccCheckResourceAttrRegionalARN(resourceName, "arn", "events", fmt.Sprintf("archive/%s", archiveName)),
+					resource.TestCheckResourceAttr(resourceName, "description", ""),
+					resource.TestCheckResourceAttr(resourceName, "event_pattern", ""),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
 func testAccAWSCloudWatchArchiveConfig(name string) string {
 	return fmt.Sprintf(`
 resource "aws_cloudwatch_event_bus" "test" {
@@ -229,6 +259,20 @@ resource "aws_cloudwatch_event_archive" "test" {
   "source": ["company.team.service"]
 }
 PATTERN
+}
+`, name)
+}
+
+func testAccAWSCloudWatchArchiveConfig_retentionOnCreation(name string) string {
+	return fmt.Sprintf(`
+resource "aws_cloudwatch_event_bus" "test" {
+  name = %[1]q
+}
+
+resource "aws_cloudwatch_event_archive" "test" {
+  name             = %[1]q
+  event_source_arn = aws_cloudwatch_event_bus.test.arn
+retention_days     = 1
 }
 `, name)
 }

--- a/aws/resource_aws_cloudwatch_event_archive_test.go
+++ b/aws/resource_aws_cloudwatch_event_archive_test.go
@@ -272,7 +272,7 @@ resource "aws_cloudwatch_event_bus" "test" {
 resource "aws_cloudwatch_event_archive" "test" {
   name             = %[1]q
   event_source_arn = aws_cloudwatch_event_bus.test.arn
-retention_days     = 1
+  retention_days   = 1
 }
 `, name)
 }

